model_analyzer/monitor/dcgm/dcgm_monitor.py

```diff
@@ -104,16 +104,18 @@ def _collect_records(self):
         for gpu in self._gpus:
             device_id = gpu.device_id()
             metrics = self.group_watcher.values[device_id]
-
             # Find the first key in the metrics dictionary to find the
             # dictionary length
             first_key = list(metrics)[0]
             num_metrics = len(metrics[first_key].values)
             for i in range(num_metrics):
                 for tag in self._tags:
                     dcgm_field = self.MODEL_ANALYZER_TO_DCGM_FIELD[tag]
+                    # DCGM timestamps are in nano seconds, Record's timestamps
+                    # are of type second.
                     records.append(
-                        tag(gpu, metrics[dcgm_field].values[i], time.time()))
+                        tag(gpu, float(metrics[dcgm_field].values[i].value),
+                            metrics[dcgm_field].values[i].ts / 1e6))
```

Comment:
This should not be converted here. Keep data as raw as possible until otherwise needed. For example, if nanoseconds is not understandable when printing to screen, then do the conversion to seconds just before printing, not as you are recording the raw data.